drivers/hmis/app/models/hmis/hud/household.rb

```diff
@@ -17,14 +17,21 @@ class Hmis::Hud::Household < Hmis::Hud::Base
   alias_attribute :household_id, :HouseholdID
 
   replace_scope :viewable_by, ->(user) do
-    viewable_households = joins(:enrollments).
-      merge(Hmis::Hud::Enrollment.viewable_by(user)). # does Data Source filter
-      pluck(:HouseholdID)
-
-    where(HouseholdID: viewable_households)
+    # correlated subquery
+    p_t = Hmis::Hud::Project.arel_table
+    hh_t = Hmis::Hud::Household.arel_table
+    projects = Hmis::Hud::Project.
+      with_access(user, :can_view_enrollment_details, :can_view_project, mode: 'all').
+      where(
+        p_t[:ProjectID].eq(hh_t[:ProjectID]).and(
+          p_t[:data_source_id].eq(hh_t[:data_source_id]),
+        ),
+      )
+    where(projects.arel.exists)
```

Comment:
Does this need a `joins(:project)` ?

Reply:
I'm probably missing something but would would joining to project do? This is using exists/correlated subuqery:
```sql
SELECT
  "hmis_households".*
FROM
  "hmis_households"
WHERE
  "hmis_households"."DateDeleted" IS NULL
  AND EXISTS (
    SELECT
      "Project".*
    FROM
      "Project"
    WHERE
      "Project"."DateDeleted" IS NULL
      AND "Project"."DateDeleted" IS NULL
      AND "Project"."id" IN (
        1,
        2,
        3,
      )
      AND "Project"."data_source_id" IS NULL
      AND "Project"."ProjectID" = "hmis_households"."ProjectID"
      AND "Project"."data_source_id" = "hmis_households"."data_source_id"
  )
```

Reply:
hopefully the `AND "Project"."data_source_id" IS NULL` isn't actually present?

Reply:
Ok, I wasn't seeing:
```
AND "Project"."ProjectID" = "hmis_households"."ProjectID"
AND "Project"."data_source_id" = "hmis_households"."data_source_id"
```
I'm good with this.  Thanks!